features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Shared/AppsAndKeys/ApiKeyManager.jsx

```diff
@@ -139,60 +178,141 @@ class ApiKeyManager extends React.Component {
-                                <FormattedMessage
-                                    id='Shared.AppsAndKeys.ViewKeys.consumer.generate.btn'
-                                    defaultMessage='Generate'
-                                />
-                            </Button>
+                          <ApiKey
+                            updateAccessTokenRequest={this.updateAccessTokenRequest}
+                            accessTokenRequest={accessTokenRequest}
+                          />
                         )}
-                        <Button onClick={this.handleClose} color='primary' autoFocus>
-                            <FormattedMessage
-                                id='Shared.AppsAndKeys.ViewKeys.consumer.close.btn'
-                                defaultMessage='Close'
-                            />
+                        {showToken && <ViewToken token={apikey} />}
+                      </DialogContentText>
+                    </DialogContent>
+                    <DialogActions>
+                      {!showToken && (
+                        <Button
+                          onClick={() => this.generateKeys(selectedApp, keyType)}
+                          disabled={!accessTokenRequest.timeout}
+                          color="primary"
+                        >
+                          <FormattedMessage
+                            id="Shared.AppsAndKeys.ViewKeys.consumer.generate.btn"
+                            defaultMessage="Generate"
+                          />
                         </Button>
+                      )}
+                      <Button onClick={this.handleClose} color="primary" autoFocus>
+                        <FormattedMessage
+                          id="Shared.AppsAndKeys.ViewKeys.consumer.close.btn"
+                          defaultMessage="Close"
+                        />
+                      </Button>
                     </DialogActions>
-                </Dialog>
+                  </Dialog>
+                </Grid>
+                {restrictSchema == "ip" && (
+                  <Grid item md={5} xs={12} style={{ padding: "20px" }}>
+                    <Card>
+                      <CardContent>
+                        <Typography variant="h5" component="h2">
+                          Examples of IP Addresses allowed
+                        </Typography>
+                        <List>
+                          <ListItem>
+                            <Typography variant="body1" component="p">
+                              Specify one IPv4 or IPv6 or a subnet using CIDR notation
+                            </Typography>
+                          </ListItem>
+                          <ListItem>
+                            <Typography variant="body1" component="p">
+                              Examples: <b>192.168.1.2</b>, <b>152.12.0.0/13</b>,{" "}
```

Comment:
Please add i18n support in this text and all the other places as well , For more info about [wso2 i18n](https://github.com/wso2/carbon-apimgt/wiki/How-internationalization-(i18n)-works-in-API-Manager-React-Apps)

Reply:
fixed